plugins/domains/scribd.com/scribd.com-error.js

```diff
@@ -1,14 +1,48 @@
+const URL = require("url");
+
 module.exports = {
 
     re: require('./scribd.com').re,
 
-    getData: function(oembedError, options, cb) {
-        if (oembedError === 401){
-            cb(null, {
-                message: "Scribd doesn't support embedding of private documents"
-            })
+    mixins: [ "*" ],
+
+    provides: ["scribdData"],
+
+    getData: function(oembedError, cb, url, options, urlMatch, request) {
+
+        if (oembedError === 401 && /secret_password=/.test(url) && urlMatch.length === 3) {
```

Comment:
You do not have `re` regex in this plugin, you link it from the main one.  Hence `urlMatch.length === 3` relies heavily on the regex in the main plugin and is error prone. Perhaps, `/secret_password=/.test(url)` is enough, especially with 401 combo?